packages/design-system/src/lib/components/file-uploader/FileUploader.tsx

```diff
@@ -0,0 +1,178 @@
+  }
+
+  const handleDrag: DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault()
+    const droppedItems = event.dataTransfer.items
+    let ok = false
+    if (droppedItems.length > 0) {
+      Array.from(droppedItems).forEach((i) => {
+        if (i.webkitGetAsEntry()?.isDirectory) {
+          ok = true
+          addFromDir(i.webkitGetAsEntry() as FileSystemDirectoryEntry)
+        } else if (i.webkitGetAsEntry()?.isFile) {
+          ok = true
+          const fse = i.webkitGetAsEntry() as FileSystemFileEntry
+          fse.file((f) => addFile(f))
+        }
+      })
+    }
+    const selectedFiles = event.dataTransfer.files
+    if (!ok && selectedFiles && selectedFiles.length > 0) {
+      addFiles(selectedFiles)
+    }
+  }
+
+  const handleRemoveFile = (f: File) => {
+    setFiles((alreadyAdded) => [...alreadyAdded].filter((x) => x !== f))
+  }
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setProgress(upload(files))
+      setFiles([...files].filter((x) => !progress.get(x.webkitRelativePath + x.name)?.removed))
+    }, 1000)
+    return () => {
+      clearInterval(interval)
+    }
+  }, [files, upload])
+
+  const inputRef = React.useRef(null)
+
+  return (
+    <Card>
+      <Card.Header>
+        <Button variant="secondary" onClick={() => (inputRef.current! as HTMLInputElement).click()}>
+          <Plus></Plus> {selectText}
+        </Button>
+      </Card.Header>
+      <Card.Body>
+        <div
+          className={styles.fileuploader}
+          onDrop={handleDrag}
+          onDragOver={(event) => event.preventDefault()}>
+          <div>
+            <input
+              ref={inputRef}
+              type="file"
+              id="filepicker"
+              onChange={handleChange}
+              multiple
+              hidden
+            />
+          </div>
+          {files.length > 0 ? (
+            <div className={styles.files}>
+              <div className={styles.group}>
+                {files
+                  .filter((x) => !progress.get(x.webkitRelativePath + x.name)?.done)
+                  .map((file) => (
+                    <>
+                      <div className={styles.file} key={file.webkitRelativePath + file.name}>
+                        <div className={styles.cell}></div>
+                        <div
+                          className={
+                            progress.get(file.webkitRelativePath + file.name)?.failed
+                              ? styles.failed
+                              : styles.cell
+                          }>
+                          {file.webkitRelativePath}
+                          {file.name}
+                        </div>
+                        <div className={styles.cell}>
+                          {progress.get(file.webkitRelativePath + file.name)?.fileSizeString}
+                        </div>
+                        <div className={styles.cell}>
+                          <ProgressBar
```

Comment:
The bar progress is going back and forward, I am testing it from a Mac on Chrome Version 125.0.6422.60 (Official Build) (arm64) if it helps.

Reply:
I made it move at random, just for testing, by assigning a progress value at random between 0 and 100. In real life, the progress should be a percentage of the bytes uploaded, making the bar only move forward.